openml/runs/run.py

```diff
@@ -191,28 +189,42 @@ def _generate_arff_dict(self):
         arff_dict['relation'] = 'openml_task_' + str(task.task_id) + '_predictions'
         return arff_dict
 
-    def _generate_trace_arff_dict(self):
-        """Generates the arff dictionary for uploading predictions to the server.
-
-        Assumes that the run has been executed.
+    def _generate_trace(self, trace_attributes, trace_content):
 
-        Returns
-        -------
-        arf_dict : dict
-            Dictionary representation of the ARFF file that will be uploaded.
-            Contains information about the optimization trace.
-        """
-        if self.trace_content is None or len(self.trace_content) == 0:
+        if trace_content is None or len(trace_content) == 0:
             raise ValueError('No trace content available.')
-        if len(self.trace_attributes) != len(self.trace_content[0]):
+        if trace_attributes is None or len(trace_attributes) != len(trace_content[0]):
```

Comment:
I meant is None. I wanted to partially controll the error, otherwise when trace_attributes would be None the len function would throw by itself. I can however handle this better